test/integration/singlecluster/tas/tas_test.go

```diff
@@ -520,6 +520,108 @@ var _ = ginkgo.Describe("Topology Aware Scheduling", ginkgo.Ordered, func() {
+				})
+
+				ginkgo.By("restart controllers", func() {
+					fwk.StopManager(ctx)
+					fwk.StartManager(ctx, cfg, managerSetup)
+				})
+
+				ginkgo.By("create wl2 which should be blocked by wl1", func() {
+					wl2 = testing.MakeWorkload("wl2", ns.Name).
+						Queue(kueue.LocalQueueName(localQueue.Name)).Request(corev1.ResourceCPU, "1").Obj()
+					wl2.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Required: ptr.To(testing.DefaultRackTopologyLevel),
+					}
+					util.MustCreate(ctx, k8sClient, wl2)
+				})
+
+				ginkgo.By("very wl2 is not admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1)
+					util.ExpectWorkloadsToBePending(ctx, k8sClient, wl2)
+					util.ExpectPendingWorkloadsMetric(clusterQueue, 0, 1)
+				})
+
+				ginkgo.By("finish wl1 and allow admitting wl2", func() {
+					util.FinishWorkloads(ctx, k8sClient, wl1)
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl2)
+				})
+			})
+
+			ginkgo.It("should respect TAS usage by admitted worklods after reboot; second workload created before reboot", func() {
+				var wl1, wl2, wl3 *kueue.Workload
+				ginkgo.By("creating wl1 which consumes the entire TAS capacity", func() {
+					wl1 = testing.MakeWorkload("wl1", ns.Name).
+						Queue(kueue.LocalQueueName(localQueue.Name)).Request(corev1.ResourceCPU, "1").Obj()
+					wl1.Spec.PodSets[0].Count = 4
+					wl1.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Preferred: ptr.To(testing.DefaultBlockTopologyLevel),
+					}
+					util.MustCreate(ctx, k8sClient, wl1)
+				})
+
+				ginkgo.By("verify wl1 is admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1)
+					util.ExpectReservingActiveWorkloadsMetric(clusterQueue, 1)
+				})
+
+				ginkgo.By("create wl2 which is blocked by wl1", func() {
+					wl2 = testing.MakeWorkload("wl2", ns.Name).
+						Queue(kueue.LocalQueueName(localQueue.Name)).Request(corev1.ResourceCPU, "1").Obj()
+					wl2.Spec.PodSets[0].TopologyRequest = &kueue.PodSetTopologyRequest{
+						Required: ptr.To(testing.DefaultRackTopologyLevel),
+					}
+					util.MustCreate(ctx, k8sClient, wl2)
+				})
+
+				ginkgo.By("very wl2 is not admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1)
+					util.ExpectWorkloadsToBePending(ctx, k8sClient, wl2)
+					util.ExpectPendingWorkloadsMetric(clusterQueue, 0, 1)
+				})
+
+				ginkgo.By("restart controllers", func() {
+					fwk.StopManager(ctx)
+					fwk.StartManager(ctx, cfg, managerSetup)
+				})
+
+				ginkgo.By("verify wl2 is still not admitted", func() {
+					util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, wl1)
+					util.ExpectPendingWorkloadsMetric(clusterQueue, 0, 1)
+				})
+
+				ginkgo.By("finish wl1 to demonstrate there is enough space for wl2", func() {
+					util.FinishWorkloads(ctx, k8sClient, wl1)
+				})
+
+				ginkgo.By("create wl3 to ensure ClusterQueue is reconciled", func() {
```

Comment:
This is a temporary workaround for another issue (outside of TAS) I noticed - if there is restart, then some of the pending workloads be classified as inadmissible and not retried until the next CQ sync. I will open a dedicated issue to fix that.

Reply:
In my testing, having more workloads made it easier to reproduce the bug. If you can still reproduce the bug with just 3 workloads please ignore my comment

Reply:
Yes, I started with your test as a basis, but it was hard to debug and track what is going on with 7 workloads. So I minimized the scenario to only 2 (or 3) workloads. This was still reproducing the issue with satisfactory rate.

Reply:
Actually, I think this issue is TAS-specific - I couldn't repro outside of TAS. I understand what is going on. 

This is because in TAS activity of the CQ may take a bit of time waiting for the tasCache. In this time it is possible that we add workload and Broadcast its presence, but the heads() function inside qManager returns no workload because it does not pass the statusChecker (as the CQ is not active yet) : https://github.com/kubernetes-sigs/kueue/blob/198076995d50e813bb60ba1a8447daeb1c2391bf/pkg/queue/manager.go#L653-L654

Removing the statusChecker fixes the problem, but it means that the scheduler may be constantly requeuing workloads belonging to not active CQs.

I will see if I can fix this as part of the PR, otherwise I would put it for future fix.

Reply:
I think this issue will be solved if we also merge https://github.com/kubernetes-sigs/kueue/pull/5309